src/client/game.cpp

```diff
@@ -3968,7 +3982,48 @@ inline void Game::updateProfilerGraphs(ProfilerGraph *graph)
 	graph->put(values);
 }
 
+/****************************************************************************
+ * Shadows
+ *****************************************************************************/
+void Game::updateShadows(float _timeoftheday)
+{
+	ShadowRenderer *shadow = RenderingEngine::get_shadow_renderer();
+	if (shadow) {
+		if (shadow->getDirectionalLightCount() == 0) {
+			shadow->addDirectionalLight();
+		}
 
+		//time of the day
+		//0 ->00:00
+		//1 -> 23_59
+		float timeoftheday = _timeoftheday - 0.2f;
+		bool isDay = false;
+		if (timeoftheday > 0.0f && timeoftheday < 0.6f)
```

Comment:
don't we have a function if we are day or night ? to create a static function in the client class (or another place if you have better) to set this flag directly. If other code parts have the same need at least we can re-use this calculation :)

Reply:
IDK if we have a function for that, I can create one in client, of course. anyway this is just to get a 0.0-x.x positive lineal range for day time to make the spherical coordinates easier.